docs/environments/gke.md

```diff
@@ -63,3 +62,74 @@ kubectl create clusterrolebinding myname-cluster-admin-binding --clusterrole=clu
 
 Once all pods are running you can return to the [User Guide](../userguide.md) to deploy the
 KubeFed control-plane.
+
+## Additional firewall rule for kubefed-admission-webhook
+
+Admission control in GKE requires an additional VPC network firewall rule.
+
+### Background
+
+The GCP Kubernetes control plane is deployed in a Google-managed account.
+
+During cluster provisioning, GKE automatically creates:
+1. A network peering connection to allow the master to communicate to the node pools.
+1. A firewall rule to allow ports 443 and 10250 from the master to the node pools.
+
+The GKE API Server operates with [`--enable-aggregator-routing=true`](https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/gci/configure-helper.sh#L1586). This means that the Service IPs are internally translated to endpoint IPs and routed to the relevant pods.
+
+A number of [webhooks are deployed](https://github.com/kubernetes-sigs/kubefed/blob/master/charts/kubefed/charts/controllermanager/templates/webhook.yaml) during the kubfed helm installation, alongside the `kubefed-admission-webhook` application. Every time a new kubefed object is created, the webhook causes the API server to query the `kubefed-admission-webhook` service.
+
+Because of the `enable-aggregator-routing` being turned on, the request is directed from the API server to the pod running the `kubefed-admission-webhook` application directly, on port 8443. This traffic needs to be allowed through the GCP firewall in order for the webhook to succeed.
+
+### Issue
+
+The kubefed-controller pod fails to start up when attempting to create a KubeFedConfig in kubernetes. A timeout error is observed.
+
+```
+Error updating KubeFedConfig "kube-federation-system/kubefed": Timeout: request did not complete within requested timeout 30s
+```
+
+### Resolution
+
+Create a VPC network firewall rule to allow traffic from your master's source IP range to the node pools. An example gcloud command and a Terraform code snippet are provided below.
+
+
+#### GCloud
+
+```bash
+gcloud compute firewall-rules create "allow-apiserver-to-kubefed-admission-webhook-8443" --allow tcp:8443 \
+      --network="[VPC_NETWORK]" \
+      --source-ranges="[MASTER_IPV4_CIDR_BLOCK]" \
```

Comment:
To go one step further, consider adding steps for the user to run CLI commands to set environment variables for these different values. For example:

```bash
VPC_NETWORK=$(gcloud ...)
MASTER_IPV4_CIDR_BLOCK=$(gcloud ...)
...
```
This would allow the user to copy/paste the commands as much as possible.